encoder.go

```diff
@@ -30,5 +33,5 @@ type encoder interface {
 	AddFields([]Field)
 	Clone() encoder
 	Free()
-	WriteEntry(io.Writer, *Entry) error
+	WriteEntry(io.Writer, string, Level, time.Time) error
```

Comment:
This undoes a change made in the previous PR - after some more thought, I think it's better to keep encoders and entries separate. This removes the concern about type mismatches between the receiver of `WriteEntry` and `Entry.encoder`.

I do still think that the `WriteEntry` name is appropriate, though, since we're writing out a single log line.